package/network/config/wifi-scripts/files/lib/netifd/hostapd.sh

```diff
@@ -729,12 +729,12 @@ hostapd_set_bss_options() {
 				return 1
 			fi
 			[ -z "$wpa_psk_file" ] && set_default wpa_psk_file /var/run/hostapd-$ifname.psk
-			[ -n "$wpa_psk_file" ] && {
+			[ -n "$wpa_psk_file" ] && [ "$auth_type" = "psk" -o "$auth_type" = "psk-sae" ] && {
 				[ -e "$wpa_psk_file" ] || touch "$wpa_psk_file"
 				append bss_conf "wpa_psk_file=$wpa_psk_file" "$N"
 			}
 			[ -z "$sae_password_file" ] && set_default sae_password_file /var/run/hostapd-$ifname.sae
-			[ -n "$sae_password_file" ] && {
+			[ -n "$sae_password_file" ] && [ "$auth_type" = "sae" -o "$auth_type" = "psk-sae" ] && {
```

Comment:
are these checks not duplicates of the ones in the previous patch ?

Reply:
they're not, the first patch will avoid creating the wpa_psk/sae_password_file files if you had a wifi-station entries defined (i.e., you had a wifi-station entry but you only have WPA2 enabled).

the second patch will avoid creating them even if you had no wifi-station entries defined.

Reply:
@rany2 you are correct, imho it makes sense to fold all 3 patches into one and annotate/explain the changes inside the patch description.
thanks alot for the effort !

Reply:
@blogic should be done :)